<!-- -->
<launch>

    <arg name="bagname" />
    <arg name="bag_start_delay" default="20.0" />

    <rosparam command="load" file="$(find carla_tools)/config/carla_ros_bridge_settings.yaml" />
    <!--param name="carla_log" value="$(find carla_tools)/carla_logs/$(arg carla_log)" /-->

    <include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
<!--         <param name="carla/synchronous_mode" value="true" /> -->
<!--         <param name="carla/fixed_delta_seconds" value="0.005" /> -->
<!--         <param name="carla/synchronous_mode_wait_for_vehicle_control_command" value="false" /> -->
    </include>

    <include file="$(find carla_ego_vehicle)/launch/carla_ego_vehicle.launch">
        <arg name="vehicle_filter" value="vehicle.*" />
        <arg name="sensor_definition_file" value="$(find carla_tools)/config/sensors.json" />
    </include>

    <node pkg="carla_tools" type="custom_manual_control.py" name="custom_manual_control" output="screen">
        <param name="role_name" value="ego_vehicle"/>
    </node>

    <node pkg="carla_tools" type="carla_to_ros_transforms.py" name="carla_to_ros_transforms" output="screen">
        <remap from="~imu/carla" to="/carla/ego_vehicle/imu" />
        <remap from="~lidar/carla" to="/carla/ego_vehicle/lidar/lidar1/point_cloud" />
    </node>

    <!--node pkg="carla_tools" type="carla_sensors.py" name="carla_sensors" output="screen">
        <param name="recording_length" value="$(arg recording_length)" />
    </node-->

    <rosparam command="load" file="$(find carla_tools)/config/carla_ros_bridge_settings.yaml" />

    <node pkg="rosbag" type="record" name="carla_rosbag_record" launch-prefix="bash -c 'sleep $(arg bag_start_delay); $0 $@' " args="
    record -O $(find carla_tools)/rosbags/$(arg bagname).bag
    /cam0/camera_info
    /cam0/image_raw
    /imu/rovio
    /imu/loam
    /imu/ros
    /lidar/carla
    /lidar/loam
    /simulation_groundtruth
    /tf
    " >
        <remap from="/cam0/camera_info" to="/carla/ego_vehicle/camera/rgb/front/camera_info" />
        <remap from="/cam0/image_raw" to="/carla/ego_vehicle/camera/rgb/front/image_color" />
        <remap from="/lidar/carla" to="/carla/ego_vehicle/lidar/lidar1/point_cloud" />
        <remap from="/lidar/loam" to="/carla_to_ros_transforms/lidar/loam" />
        <remap from="/imu/rovio" to="/carla_to_ros_transforms/imu/rovio" />
        <remap from="/imu/loam" to="/carla_to_ros_transforms/imu/loam" />
        <remap from="/imu/ros" to="/carla_to_ros_transforms/imu/ros" />
        <remap from="/simulation_groundtruth" to="/carla/ego_vehicle/vehicle_status" />
    </node>
    
</launch>
